src/common/types.ts

```diff
@@ -5,7 +5,7 @@ import EventBus from '../EventBus';
 
 export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
 
-export interface Column<TRow> {
+export interface Column<TRow, TSummaryRow> {
```

Comment:
```suggestion
export interface Column<TRow, TSummaryRow = never> {
```
That way we can remove a bunch of `never` in `Column<..., never>`